cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -528,20 +505,21 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 			a.lastScaleUpTime.Add(a.ScaleDownDelayAfterAdd).After(currentTime) ||
 			a.lastScaleDownFailTime.Add(a.ScaleDownDelayAfterFailure).After(currentTime) ||
 			a.lastScaleDownDeleteTime.Add(a.ScaleDownDelayAfterDelete).After(currentTime)
+		deletionsInProgress := len(a.scaleDownActuator.DeletionsInProgress())
 		// In dry run only utilization is updated
-		calculateUnneededOnly := scaleDownInCooldown || scaleDown.IsNonEmptyNodeDeleteInProgress()
+		calculateUnneededOnly := scaleDownInCooldown || deletionsInProgress > 0
```

Comment:
Presumably the deletionsInProgress > 0 check is something that we'll actually need to push into legacy implementation? I'm fine if the answer is "yes, in a future PR" :)